### ISSUE TITLE
Reset Focus After Enter/Space Selection in DatePicker Popup Mode To Original Input

### ISSUE THREAD

**Issue by acgentry** [angular-ui/bootstrap#4243]
In the datepicker, in popup mode, if you select a date using the enter/space key the original input box is no longer in focus.  This is reproducible in the directives examples [page](https://angular-ui.github.io/bootstrap/#datepicker).

I am unable to see any settings or hooks that would fix this behavior.  Pardon if I am missing something.

fyi, I am aware that I can create a directive to set focus back to the input, but I would assume this would be handled better within the datepicker directive itself.

acg

**Comment by krisurnok** [comment]
https://github.com/angular-ui/bootstrap/issues/4244

**Comment by krisurnok** [comment]
i have made some change in code. please try it
https://github.com/angular-ui/bootstrap/issues/4244

**Comment by juanpujol** [comment]
:+1: Please I'll also appreciate a patch for this bug. Thank you.

**Comment by stereocilia** [comment]
+1

-Problem produced in plunkr (with some details): http://plnkr.co/edit/QVrbHA?p=preview
-My PR: https://github.com/angular-ui/bootstrap/pull/4409
-Build from my PR in plunkr: http://plnkr.co/edit/glvMLw?p=preview

I'm not sure if this is the best solution, but it seems to work in FF, Chrome, IE11.

**Comment by juanpujol** [comment]
thank you @SteffanLong, is Safari I noticed that the focus doesn't work on the calendar icon. But it's way better than right now. Thanks again.

**Comment by wesleycho** [comment]
Ah shoot, I didn't realize that there was an issue attached to the PR - this should now be addressed by https://github.com/angular-ui/bootstrap/commit/36ecf60c234454af59f4d1d1e6c40bea7e4ea96c

**Comment by acgentry** [comment]
Thanks for doing the heavy lifting @stereocilia
